src/dag-node-factory.js

```diff
@@ -0,0 +1,139 @@
+'use strict'
+
+const stable = require('stable')
+const sort = stable.inplace
+const mh = require('multihashes')
+const util = require('./util')
+const DAGLink = require('./dag-link')
+
+class DAGNodeFactory {
+  static create (data, dagLinks, hashAlg) {
```

Comment:
Also, I can see the value of having the create function in a separate file, having to do: `dagPB.dagNodeFactory.create(..)`, seems a tad long, just to create an instance, we can scope the `create` function withing dagPB.DAGNode even if it is in a separate file.